test/e2e/clusterclass_rollout.go

```diff
@@ -590,6 +624,119 @@ func assertMachineDeployments(g Gomega, clusterClassObjects clusterClassObjects,
 	}
 }
 
+func assertMachinePools(g Gomega, clusterClassObjects clusterClassObjects, clusterObjects clusterObjects, cluster *clusterv1.Cluster, clusterClass *clusterv1.ClusterClass) {
+	for _, machinePool := range clusterObjects.MachinePools {
+		mpTopology := getMPTopology(cluster, machinePool)
+		mpClass := getMPClass(cluster, clusterClass, machinePool)
+
+		// MachinePool.metadata
+		g.Expect(machinePool.Labels).To(BeEquivalentTo(
+			union(
+				map[string]string{
+					clusterv1.ClusterNameLabel:                    cluster.Name,
+					clusterv1.ClusterTopologyOwnedLabel:           "",
+					clusterv1.ClusterTopologyMachinePoolNameLabel: mpTopology.Name,
+				},
+				mpTopology.Metadata.Labels,
+				mpClass.Template.Metadata.Labels,
+			),
+		))
+		g.Expect(
+			union(
+				machinePool.Annotations,
+			).without(g, clusterv1.RevisionAnnotation),
+		).To(BeEquivalentTo(
+			union(
+				mpTopology.Metadata.Annotations,
+				mpClass.Template.Metadata.Annotations,
+			),
+		))
+
+		// MachinePool.spec.template.metadata
+		g.Expect(machinePool.Spec.Template.Labels).To(BeEquivalentTo(
+			union(
+				map[string]string{
+					clusterv1.ClusterNameLabel:                    cluster.Name,
+					clusterv1.ClusterTopologyOwnedLabel:           "",
+					clusterv1.ClusterTopologyMachinePoolNameLabel: mpTopology.Name,
+				},
+				mpTopology.Metadata.Labels,
+				mpClass.Template.Metadata.Labels,
+			),
+		))
+		g.Expect(machinePool.Spec.Template.Annotations).To(BeEquivalentTo(
+			union(
+				mpTopology.Metadata.Annotations,
+				mpClass.Template.Metadata.Annotations,
+			),
+		))
+
+		// MachinePool InfrastructureMachineTemplate.metadata
+		ccInfrastructureMachineTemplate := clusterClassObjects.InfrastructureMachinePoolTemplateByMachinePoolClass[mpClass.Class]
+		ccInfrastructureMachineTemplateTemplateMetadata := mustMetadata(contract.InfrastructureMachineTemplate().Template().Metadata().Get(ccInfrastructureMachineTemplate))
```

Comment:
contract.InfrastructureMachineTemplate should be contract.InfrastructureMachinePoolTemplate

(similar below)